### ISSUE TITLE
Instances killed due to full filesystem (most likely - logs)

### ISSUE THREAD

**Issue by bocytko** [zalando-stups/taupage#176]
The `/var/log` is mounted on the root partition. This way, it's easy to kill an instance by writing too many logs or due to bugs in the application/3rd party libraries.

Proper log rotation in the applications and in the Taupage, plus resolving:
- #142 
- #147 
- #172 

does not really solve the main issue - having a possibility to kill instances by having a full filesystem.

File systems should be better separated from each other to achieve higher instance reliability. Separating logs & maybe the home directory could be a good step in this direction.

```
$ df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/xvda1      7.8G  4.1G  3.3G  56% /
none            4.0K     0  4.0K   0% /sys/fs/cgroup
udev            1.9G   12K  1.9G   1% /dev
tmpfs           377M  380K  377M   1% /run
none            5.0M     0  5.0M   0% /run/lock
none            1.9G   76K  1.9G   1% /run/shm
none            100M     0  100M   0% /run/user
```

**Comment by femueller** [comment]
I don't think it makes sense to store the logs in the home directory. It's not designed to be a log storage by the Ubuntu/Debian guidelines.
However, I agree that this is an EC2 issue. It's because changing the partitioning scheme is something which is really hard to achieve.
From my perspective a way cleaner and more scalable solution for people who don't know how their /var/ storage will grow, they could just mount an EBS volume in that location:
https://docs.stups.io/en/latest/components/taupage.html#ebs

**Comment by bocytko** [comment]
> I don't think it makes sense to store the logs in the home directory

It's not what I suggested. I asked for a separation between:

- `/`
- `/var/log`
- `/home` (not really needed, but imagine you ssh to an instance and you create a big file in $HOME)

**Comment by apfeiffer85** [comment]
Thats not easy doable because the "user/developer" is able to choose a custom size for the whole HDD and could also add more HDD (e.g. EBS volumes). 
So for example a user decide to only use a 5GB HDD than we decide to split this up to maybe: 

/ - 1 GB
/var/log - 2 GB
/home/ - 2 GB 

And then maybe / is even not big enough for the application it self ... so i guess we could not find a really good solution to handle this. 

We should better take care that there are as few logs as necessary on the EC2 instance / taupage and if there are logs they should be always rotated and deleted regularly. Therefore we can prevent full filesystem regarding logs .... ;-)

**Comment by bocytko** [comment]
I understand your point @apfeiffer85.

Found another way to exploit this, even when #172 is set. Even if you rotate on size, the logrotation is scheduled by cron hourly. This way, you could fill the filesystem before the logrotation kicks in :)

So basically the rotation on size should be considered more often than hourly.

**Comment by bocytko** [comment]
We see instances becoming zombies if the disks are full. Especially OAuth credential rotation via `berry` fails with a full disk, eventually causing a 100% error rate after the disk itself is full for some time.

Any suggestions? I'd expect an instance to work correctly incl. audit logs and credential rotation even if the volume containing application logs is full.

**Comment by sezera** [comment]
" Even if you rotate on size, the logrotation is scheduled by cron hourly. This way, you could fill the filesystem before the logrotation kicks in :)"

This affected many teams, who had to increase disk size as workaround.

**Comment by sezera** [comment]
Even if you rotate on size, the logrotation is scheduled by cron hourly. This way, you could fill the filesystem before the logrotation kicks in :)

**Comment by lhotari** [comment]
There is a problem that logrotate might fail when the disk is full. More details in https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=693985 , that's fixed by https://github.com/logrotate/logrotate/pull/16 and included in logrotate release 3.10.0 . It might take some time to get the fix to Ubuntu.
